homeassistant/components/universal/media_player.py

```diff
@@ -148,8 +148,9 @@ def async_on_dependency_update(*_):
         for entity in self._attrs.values():
             depend.append(entity[0])
         if self._state_template is not None:
-            for entity in self._state_template.extract_entities():
-                depend.append(entity)
+            self.hass.helpers.event.async_track_template_result(
```

Comment:
Pass the unsubscribe function to `self.async_on_remove` (do same for track state change event below)